operator/src/main/java/oracle/kubernetes/utils/OperatorUtils.java

```diff
@@ -19,4 +19,62 @@ public static String joinListGrammatically(final List<String> list) {
             .concat(list.get(list.size() - 1))
         : list.get(0);
   }
+
+  /**
+   * Compare the 'numero lexi sorting name' as defined in {@link #getSortingString(String)} of the
+   * given 2 Strings.
+   *
+   * @param str1 First string for comparison
+   * @param str2 Second string for comparison
+   * @return a negative integer, zero, or a positive integer as the sorting name of str1 * is less
+   *     than, equal to, or greater than the sorting name of str2.
+   */
+  public static int compareSortingStrings(String str1, String str2) {
+    if (str1 == null || str2 == null) {
+      if (str2 != null) {
+        return -1;
+      } else if (str1 != null) {
+        return 1;
+      }
+      return 0;
+    }
+    return getSortingString(str1).compareTo(getSortingString(str2));
+  }
+
+  /**
+   * The 'numero lexi sorting name' is a munged version of a
+   * string that 'zero fills' its numeric portions. This can
+   * be used to ensure "member2foo" sorts before "member12foo"
+   * as these would munge to "member000002foo" and "member000012foo".
+   *
+   * <p>Handles up to 6 digits... - otherwise it doesn't zero fill...
+   */
+  public static String getSortingString(String orig) {
+    String ret = "";
+    String word = "";
+    char lastCh = 0;
+    for (char ch : orig.toCharArray()) {
+      if (word.length() != 0
+          && Character.isDigit(ch) ^ Character.isDigit(lastCh)) {
+        ret += getSortingWord(word);
+        word = "";
+      }
+      word += ch;
+      lastCh = ch;
+    }
+    ret += getSortingWord(word);
+    return ret;
+  }
+
+  private static String getSortingWord(String word) {
+    if (word.length() == 0) {
+      return word;
+    }
+    if (Character.isDigit(word.charAt(0))) {
+      for (int i = word.length(); i < 6; i++) {
```

Comment:
How about something more generous than 6 - like 20?  The numbers we're padding come from the customer and there's not much need to be stingy here.

Reply:
done. Yes, we should be more generous.